rs/crypto/secp256k1/src/lib.rs

```diff
@@ -833,25 +824,18 @@ impl PublicKey {
 
     /// Verify a BIP340 (message,signature) pair
     pub fn verify_bip340_signature(&self, message: &[u8], signature: &[u8]) -> bool {
-        use k256::elliptic_curve::point::AffineCoordinates;
         use k256::schnorr::signature::hazmat::PrehashVerifier;
-        use std::ops::Neg;
 
         let signature = match k256::schnorr::Signature::try_from(signature) {
             Ok(sig) => sig,
             Err(_) => return false,
         };
 
-        let pt = self.key.to_projective().to_affine();
+        let pt = self.serialize_sec1(true);
 
-        let pt = if pt.y_is_odd().into() { pt.neg() } else { pt };
-
-        if let Ok(pk) = k256::PublicKey::from_affine(pt) {
-            if let Ok(bip340) = k256::schnorr::VerifyingKey::try_from(pk) {
-                bip340.verify_prehash(message, &signature).is_ok()
-            } else {
-                false
-            }
+        // from_bytes takes just the x coordinate encoding:
+        if let Ok(bip340) = k256::schnorr::VerifyingKey::from_bytes(&pt[1..]) {
```

Comment:
Would `if let Ok(bip340) = k256::schnorr::VerifyingKey::try_from(&self.key) {` be an option here?

Update: interestingly, with this, `should_accept_bip340_signatures_that_we_generate` consistently fails, but `should_match_bip340_reference_test_signatures` succeeds.

Reply:
Due to some quirk in the API, `try_from` fails if the point has an odd `y`. https://github.com/RustCrypto/elliptic-curves/issues/1070#issuecomment-2357237557